### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by undef1nd**

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -69,10 +69,40 @@ pub enum Header {
     Item(Item),
 }
 
+pub enum HType {
+    L,
+    D,
+    I,
```
Comment (valenting) [inline src/parser.rs:75]: You might want to be a little more verbose here.

File: src/parser.rs [hunk 2]
```diff
@@ -69,10 +69,40 @@ pub enum Header {
...
+    pub fn parse_dict_header(input_bytes: &[u8]) -> ParserResult<Dictionary> {
+        if let Header::Dictionary(dict) = Self::parse(input_bytes, HType::D)? {
+            Ok(dict)
+        } else {
+            Err("not a dictionary header")
+        }
+    }
+
+    pub fn parse_list_header(input_bytes: &[u8]) -> ParserResult<List> {
+        if let Header::List(dict) = Self::parse(input_bytes, HType::L)? {
```
Comment (valenting) [inline src/parser.rs:90]: maybe not name them dict. 🙂

File: src/parser.rs [hunk 3]
```diff
@@ -69,10 +69,40 @@ pub enum Header {
...
+            Ok(dict)
+        } else {
+            Err("not a list header")
+        }
+    }
+
+    pub fn parse_item_header(input_bytes: &[u8]) -> ParserResult<Item> {
+        if let Header::Item(dict) = Self::parse(input_bytes, HType::I)? {
```
Comment (valenting) [inline src/parser.rs:98]: here too

File: src/parser.rs (hunk 2 above)
Comment (undef1nd) [reply src/parser.rs:90]: Very good comment)) But I'm not happy with the whole change, to be honest

File: src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,65 @@
...
+use indexmap::IndexMap;
+pub use rust_decimal::Decimal;
+
 pub mod parser;
 pub mod serializer;
-mod utils;
+pub(crate) mod test_parser;
+pub(crate) mod test_serializer;
+
+type Res<R> = Result<R, &'static str>;
```
Comment (valenting) [inline src/lib.rs:10]: nit: I think it's common to call it `Result<T>` instead of `Res<T>`

**Pull request merged**